lib/src/equatable_utils.dart

```diff
@@ -13,21 +13,7 @@ bool equals(List<Object?> a, List<Object?> b) {
   if (identical(a, b)) return true;
   if (a.length != b.length) return false;
   for (var i = 0; i < a.length; i++) {
-    final unitA = a[i];
-    final unitB = b[i];
-    if (_isEquatable(unitA) && _isEquatable(unitB)) {
-      return unitA == unitB;
```

Comment:
This implementation is incorrect since this will incorrectly return true when the first two objects are equal in a list of props (even if the rest aren't):

```dart
class Person with EquatableMixin {
  Person({required this.name});

  final String name;

  @override
  List<Object?> get props => [name];
}

test('...', () {
    final alice = Person(name: 'Alice');      
    expect(equals([alice, null], [alice, -1]), isFalse);
});
```

The above test incorrectly fails.